packages/utilities/exposure-scanning/scan/index.js

```diff
@@ -0,0 +1,77 @@
+const replacementStr = '*';
+
+export const scanText = (text, regex) => {
+  let foundTextMatch = false;
+  const scannedText = text.replace(regex, replacementStr);
+
+  if (scannedText !== text) {
+    foundTextMatch = true;
+  }
+
+  return { body: scannedText, foundTextMatch };
+};
+
+export const scanComments = (comments, regex) => {
+  let foundCommentMatch = false;
+  const scannedComments = [];
+
+  comments.forEach(({ id, body }) => {
+    if (regex.test(body)) {
+      foundCommentMatch = true;
+      scannedComments.push({
+        id,
+        body: body.replace(regex, replacementStr),
+      });
+    }
+  });
+
+  return { comments: scannedComments, foundCommentMatch };
+};
+
+export const scanPr = (pr, regex) => {
+  let scannedPr = pr;
+  let foundMatch = false;
+
+  try {
+    const { body, foundTextMatch } = scanText(pr.body, regex);
+    const { comments, foundCommentMatch } = scanComments(pr.comments, regex);
+    const {
+      comments: reviewComments,
+      foundCommentMatch: foundReviewCommentsMatch,
+    } = scanComments(pr.reviewComments, regex);
+
+    scannedPr = {
+      body,
+      comments,
+      reviewComments,
+    };
+
+    foundMatch =
+      foundTextMatch || foundCommentMatch || foundReviewCommentsMatch;
+  } catch (error) {
+    throw new Error('Encountered an error when scanning.');
+  }
+
+  return { ...scannedPr, foundMatch };
```

Comment:
It's easier to read this code if `foundMatch`is explicitly set to false rather than hunting down the variable definition and realising it's mutable and hunting down where it might mutate and then realising it would only ever be false here

```suggestion
  return { ...scannedPr, foundMatch: false };
```